### PR TITLE
Change local package naming to recognized name

### PR DESCRIPTION
**Pull request by aweimeow**
This change have modify package's naming from haslib's hash to
meaningful naming, it will grab name from target project's setup.py.

### CODE REVIEW

**Comment by uranusjr** [comment]
I wonder if it is a good idea to fall back to the old hash method if `check_output` fails…

**Comment by techalchemy** [comment]
Thanks for the contribution! We definitely want to fall back to the old hash method if we can't get the name from the setup.py

File: src/requirementslib/requirements.py [hunk 1]
```diff
@@ -295,13 +295,14 @@ def get_uri(self):
...
     def get_name(self):
+        from subprocess import check_output, PIPE
         loc = self.path or self.uri
         if loc:
             self._uri_scheme = "path" if self.path else "uri"
-        hashed_loc = hashlib.sha256(loc.encode("utf-8")).hexdigest()
-        hash_fragment = hashed_loc[-7:]
-        self._has_hashed_name = True
-        return hash_fragment
+        name = check_output([sys.executable, "setup.py", "--name"], cwd=loc).strip()
```
[the remark this answers is not in the document]
Comment (uranusjr) [reply src/requirementslib/requirements.py]: Ah I didn’t know about `run_setup`. This is nice.

[the remark this answers is not in the document]
Comment (techalchemy) [reply src/requirementslib/requirements.py]: I included the commit(s) here and built on it so I can get this moving, thanks again!

**Pull request closed without merging**